### PR TITLE
Inherit datasources from Nest

### PR DESCRIPTION
**Pull request by metlos**
This builds on top of 9fb1beb1a6ec902476ff0876c522670eef8c9715 (PR #128) and changes the way KeycloakDS is defined.

We used to just inherit the datasources from keycloak, now we inherit the  datasources from nest and add keycloak-specific DS to those.

NOTE: this depends on https://github.com/hawkular/hawkular-commons/pull/44

### CODE REVIEW

**Comment by jpkrohling** [comment]
Agree, we should treat `KeycloakDS` as an "implementation detail" of the FP we consume. Unless we don't have a choice.

**Comment by metlos** [comment]
IMHO both datasources are demo ones - using H2 is not recommendable for production, so users will have to change those anyway.

I'm not sure which workflow is better either:
- Define KeycloakDS ourselves
  1. Update KC
  2. Find out the data source doesn't work anymore
  3. Update it so that it does on one place
- Define InventoryDS ourselves
  1. Inventory DS needs to change, people go to nest to update it, because that's where inventory is deployed to.
  2. But it still doesn't work.
  3. Oh, it needs to be updated in accounts, too.

Note that integrating with KC is solely the responsibility of accounts, while inventory ends up in a couple of places. While I'm not too fussed about the second approach, I'd like to avoid the "oh" moments as much as possible.

**Comment by metlos** [comment]
The problem we're dealing here is that accounts inherits two FPs - keycloak and nest. Because multiple inheritance is not clearly supported one has to solve "conflicts" manually, which is what we're trying to do here. Either of the DS's can be applied manually. 

All I am saying is that nest is the better one of the two to inherit w/o changes because it is used on many places in our codebase. KC FP is inherited only in accounts. So inherting nest as is lowers the potential for surprises for integrators putting nest, accounts and the rest of hawkular together. 

As soon as accounts and nest FPs diverge, the whole hawkular has a problem. If KC and accounts diverge, the problem is isolated to accounts.

But I'll stop my rant here. If you find my points moot, I will update the PR to redefine inventory ds instead. Let me know.

**Comment by metlos** [comment]
Btw. this is failing because hawkular/hawkular-commons#44 hasn't been merged yet.

**Comment by ppalaga** [comment]
> Define InventoryDS ourselves
> Inventory DS needs to change, people go to nest to update it, because that's where inventory is deployed to.
> But it still doesn't work.
> Oh, it needs to be updated in accounts, too.

@metlos from the very beginning I said I prefer defining the InventoryDS here in Accounts and not defining it Nest.
As you probably know:

(1) Accounts FP depends both on Nest FP and KC FP and

(2) Accounts FP is the only component in the universe consuming Nest FP directly. All Hawkular mid and high level components (Inv, GW, Alerts, Metrics, Agent, HK main) consume Accounts FP rather than Nest FP directly.

From the maintenance PoV, the best place to define the InventoryDS is here in Accounts. 

An architecturaly cleaner alternative would be to define it /somehow/ in Inventory. XSL applied on the whole standalone.xml is one possible but complicated (because the same XSL needs to be maintained in the whole downstream of Inventory) way how to do that. Creating an inventory FP does not make much sense, IMO.

To sum up: I vote for hard-coding the InventoryDS to the XSL file that transforms KC's datasources.xml here in Accounts and I vote for rejecting https://github.com/hawkular/hawkular-commons/pull/44

**Comment by metlos** [comment]
@ppalaga, yes we've been through this privately already and we couldn't come to agreement ;) Let's just let @jpkrohling decide whether he wants to "pollute" accounts code with something strictly required by inventory or whether he's rather willing to go through some hoops (redefine KC DS in accounts) to make it isolated from inventory.

**Comment by jpkrohling** [comment]
Honestly, I prefer to have KeycloakDS in Accounts than InventoryDS, if that's the choice to be made. If there is a way to not have any, then that's optimal. But if one of them have to be defined here, I'd say that Keycloak is the preferred option.

**Comment by metlos** [comment]
@ppalaga do you have an idea how to "merge" the two datasource xmls? I completely agree with @jpkrohling that the optimal solution is not to redefine anything in accounts and just inherit, but I fail to see a way of doing that. But you are our XSL and FP mage.. ;)

**Comment by jpkrohling** [comment]
Can this one be closed? I think it's now outdated, right?

**Comment by ppalaga** [comment]
@jpkrohling yes, can be closed. @metlos I have indeed found a way how to merge multiple xml files using XSL - see https://github.com/hawkular/hawkular-accounts/pull/137
